### PR TITLE
autocomplete support!

### PR DESCRIPTION
**Pull request by HermesCheng**
1. support autocomplete and tag
2. support 'modifier:search string'
3. space support @2
4. fix suite edit page wrap twice bugs
5. add some css for #searchinput

### CODE REVIEW

File: js/textext.plugin.moztrap-ajax.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+				url     : url,                              //point to moztrap API url
+				data    : dataCallback(query),
+			        //TODO: support different response type
+				success : function(data) { self.onComplete(data, query) },
+				error   : function(jqXHR, message) { console.error(message, query) }
+			}, 
+			self.opts('ajax')
+		);
+	        //TODO: initiate multiple call to all REST APIs
+		$.jsonp(opts); //TODO: maybe change this with jsonp?
```
Comment (shinglyu) [inline js/textext.plugin.moztrap-ajax.js]: We have already removed MozTrap API's CORS restrictions. I guess we can just use `$.ajax()` instead.

**Comment by HermesCheng** [comment]
@shinglyu , I have updated the commit.
1. support autocomplte and tag
2. support 'modifier:search string'
3. space support @2
4. support user can input "several 'tagged' queries + 'non-tagged' query"
5. fix suite edit page wrap twice bugs
6. add some css for #searchinput

KNOWN ISSUES: only get limited results (20), and does not update if user input is not in there

**Comment by shinglyu** [comment]
Great work! I'll review them on Mon.

**Comment by HermesCheng** [comment]
@shinglyu 
I have updated the commit.

**Comment by shinglyu** [comment]
Oops, there are conflicts because of a patch landed previously. Could you kindly pull it and resolve the conflict, then push it again? Should be fairly trivial conflict.

**Comment by HermesCheng** [comment]
@shinglyu , I have rebased it. Could you try again?

**Pull request merged**